### PR TITLE
Stop running onRemove during resize

### PR DESCRIPTION
**Pull request by luanpotter**
# Description

Stop running onRemove during onResize (which caused `parent` to be nulled out, causing several issues)

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [ ] My PR includes unit or integration tests for *all* changed/updated/fixed behaviors and are passing (See [Contributor Guide]).
- [x] My PR does not decrease the code coverage, or I have __a very special case__ and explained on the PR description why this PR decreases the coverage.
- [x] I updated/added relevant documentation (doc comments with `///`) and updated/added examples in `doc/examples`.
- [x] I have formatted my code with `./scripts/format.sh` and the Flame analyzer (`./scripts/analyze.sh`) does not report any problems.
- [x] I read and followed the [Flame Style Guide].
- [ ] I have added a description of the change under `[next]` in `CHANGELOG.md`.
- [x] I removed the `Draft` status, by clicking on the `Ready for review` button in this PR.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require Flame users to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change (please indicate a breaking change in `CHANGELOG.md`).
- [x] No, this is *not* a breaking change.

## Related Issues

https://github.com/flame-engine/flame/issues/967
https://github.com/flame-engine/flame/issues/977

[issue database]: https://github.com/flame-engine/flame/issues
[Contributor Guide]: https://github.com/flame-engine/flame/blob/main/CONTRIBUTING.md
[Flame Style Guide]: https://github.com/flame-engine/flame/blob/main/STYLEGUIDE.md

### CODE REVIEW

**Review by erickzanardo — APPROVED** [review]
LGTM, if we could test this somehow, would be awesome, so we don't fall on the same bug in the future.

**Comment by luanpotter** [comment]
i'm not sure how to cause a resize on the widget level test, but I can investigate

**Comment by erickzanardo** [comment]
> i'm not sure how to cause a resize on the widget level test, but I can investigate

we could put the game widget inside a container, and in the middle of the test, change the size of the container. Not sure if it will work, but is something I would try

**Comment by luanpotter** [comment]
@erickzanardo great idea, it worked

File: packages/flame/test/game/game_widget/game_widget_lifecycle_test.dart [hunk 1]
```diff
@@ -158,6 +185,20 @@ void main() {
       );
     });
 
+    testWidgets('on resize, parents are kept', (tester) async {
+      final events = <String>[];
+      await tester.pumpWidget(MyContainer(events));
+
+      events.clear();
+      final state = tester.state<MyContainerState>(find.byType(MyContainer));
+      state.causeResize();
```
Comment (erickzanardo) [inline packages/flame/test/game/game_widget/game_widget_lifecycle_test.dart:194]: TIL

**Review by erickzanardo — APPROVED** [review]

**Pull request merged**